lib/gis/ls_filter.c

```diff
@@ -144,14 +147,36 @@ static int wc2regex(struct buffer *buf, const char *pat)
 
 static int re_filter(const char *filename, void *closure)
 {
+#ifdef HAVE_REGEX_H
     regex_t *regex = closure;
+	return filename[0] != '.' && regexec(regex, filename, 0, NULL, 0) == 0;
+#endif
+#ifdef HAVE_PCRE_H
+	const char* pcreErrorStr;
+	pcre_extra* pcreExtra;
+	int pcreExecRet;
+	pcre* pcre_regex = closure;
+	// Optimize the regex
```

Comment:
```suggestion
	/* Optimize the regex */
```

(please use C style comments)